### ISSUE TITLE
GXAmplifier-X & GXAmplifier-Stereo-X are also affected by issue #82

### ISSUE THREAD

**Issue by Gimmeapill** [brummer10/guitarix#86]
Hi Hermann,

[Issue #82  ](https://github.com/brummer10/guitarix/issues/82) is also affecting the LV2 versions GXAmplifier-X & GXAmplifier-Stereo-X. (Since those are unlikely to be used for bass, re-enabling the high pass filter shouldn't be too problematic).

Br,

LX

**Comment by Rippert** [comment]
Actually, I have used those for Bass. I know I'm a rare breed (pronounced weird). They are very useful when using a plugin host like Carla.

I think that there is more to this than just the highpass filter. There really isn't a good reason for a simple filter (or the lack of one) to cause the relatively complex behavior that you noted originally in #82. While the switch will work, really there should be a simpler, universal cure for this that allows both Bass and high gain Guitar to be played through the amp. When I get some more time, I will look at the code in detail and see if there is a simpler way to get this to work. First, I'll have to reproduce the issue you noted.

Would you mind if I used this issue to correspond with you, and Hermann or others? The old issue is closed. I could also open a new issue if that is preferable.

**Comment by Gimmeapill** [comment]
Sure, no worries. Let's use this issue if you feel like digging deeper (although I won't be able to help much beyond A/B testing).
I am also using Guitarix occasionally with a 5 string bass and I actually like the highpass filter - it does help saving my desktop studio monitors. But I can understand that you'd be missing some rumble with a beefier setup ;-)
This bug set aside, wouldn't you be better off just running it in parallel with a bass amp (a la Lemmy)? 
I don't think you could expect a HW high gain guitar amp to go that low, so 31Hz with a switch to go lower is already quite fair IMO...

**Comment by Rippert** [comment]
A/B testing would be great.

I mostly use Guitarix for live situations with my main Bass amp (2x15 sealed cab and/or a 15 + 10 plus Horn Triax from SWR), so I do have a beefier setup. I actually have been using it in parallel with a clean signal through my mixer up until I convinced Hermann to drop the HPF. I actually asked for a switch to bypass it like the current code, but Hermann couldn't remember why he had the HPF there and it didn't seem to hurt the sound to remove it.

With the HPF gone I can get a much tighter and deeper sound than I can by running in parallel. Both setups have a place and a useful tone, but I just couldn't get the old fashioned 60s-70s amp tone, like I got out of my old Bassman 100, with a parallel setup. That old amp is nice, but it's just too expensive to keep running with today's low quality power tubes.

These HPF setups seem to be ubiquitous in digital code/gear. Even back in the 90s when I used a Digitech GSP2101 for my Bass rig, it would lose a lot of bottom end whenever I took it off bypass to use the analog distortion or the digital section. I would have to use a fairly extreme EQ setup to get the bottom back, but it always sounded muddy. So I'm very happy Hermann was willing to drop the HPF, but I certainly don't want to hurt anyone else's tone, so the switch is OK. I'm more curious about why it doesn't seem to work like a real tube amp that can put out high gain tone without losing all the low end.

Also, you might note that there is not just a 31 Hz filter on the single tubes, but a 310 Hz filter on the all the Push-Pull power tube setups, which really kills the bass tone.  Not really sure why they have such a high cutoff, but that also didn't seem to hurt anything by removing it at the time. That's also switched in and out with the current code if I'm reading Hermann's commit right.

**Comment by brummer10** [comment]
> These HPF setups seem to be ubiquitous in digital code/gear. 

Yes, because digital distortion produce overtones, this mean that the low end will be move over to the mid range. That makes the sound muddy for guitar. 

> Also, you might note that there is not just a 31 Hz filter on the single tubes, but a 310 Hz filter on the all the Push-Pull power tube setups

As far I know there isn't a 310Hz high-pass on the Push-Pull sections. 

Still. I'm not sure how to handle this issue. Adding a switch to select between "High Gain" and "Low end" will be the most simple solution, just that it may disturb existing sessions hold me back to do it.

**Comment by Gimmeapill** [comment]
Well, if high gain guitar & bass cannot work at the same time, this sounds more like an architecture choice to make at this stage. 
I'm absolutely fine with the switch, but still think that the original scope of those amps being the guitar, the nice to have features shouldn't affect the original functionality (and the opposite for the bass amps). 
New users might get a bit confused also, since this is not the typical control you would find on a HW amp - the closest thing that comes to mind would be the clean/crunch channels selectors.
So maybe presenting the switch as a channel selector labelled something like guitar/bass would be more intuitive?

**Comment by Rippert** [comment]
Yeah, it's actually the midrange that you want for a bass amp so that it "cuts through the mix" when you give it some tube distortion. You also get a really nice sustain/compression on the deep sub harmonics from an overdriven tube amp on the bass.

I think presenting the switch as a Guitar/Bass channel selector is a good idea, with Guitar being the default so older presets don't change by default.

I think you'll find that some guitarists prefer the Bass setting just like many guitarists like to use the old Fender Bassman head. And there may be bassists that prefer the Guitar setting too.

I looked back at the 310 Hz filter and it was actually a speaker emulator filter on the push-pull simulation that got removed a while back and wasn't changed in the commit relative to this issue.

**Comment by fernando-inf** [comment]
Hi, I don't know if you noticed, but LV2 GXMetalHead is also corrupted. It only works well if you put all the potentiometers to the maximum, I don't usually use these LV2 plugins but I warn you just in case.

I have tested it with `~$ jalv.gtk3 http://guitarix.sourceforge.net/plugins/gxmetal_head `

**Comment by brummer10** [comment]
I've not forgotten about this, still working on it.

**Comment by brummer10** [comment]
So, it's done. I added a new switch to the GxAmplifiers to select between guitar or bass input to use the according filter. 
Also @fernando-inf , I've fixed GxMetalHead, but that was a different issue.

**Comment by Rippert** [comment]
Tested both LV2 GxAmp plugins today. The new switch works, and it looks good in the GUI. Thanks Hermann.

**Comment by Gimmeapill** [comment]
Confirmed as well - the switch works fine (and looks good), thanks again Hermann!
